pkg/virt-launcher/virtwrap/network/podinterface_test.go

```diff
@@ -636,31 +633,6 @@ var _ = Describe("Pod Network", func() {
 			})
 		})
 	})
-
-	It("should write interface to cache file", func() {
-		uid := types.UID("test-1234")
-		vmi := &v1.VirtualMachineInstance{ObjectMeta: v12.ObjectMeta{UID: uid}}
-		address1 := &net.IPNet{IP: net.IPv4(1, 2, 3, 4)}
-		address2 := &net.IPNet{IP: net.IPv4(169, 254, 0, 0)}
-		fakeAddr1 := netlink.Addr{IPNet: address1}
-		fakeAddr2 := netlink.Addr{IPNet: address2}
-		addrList := []netlink.Addr{fakeAddr1, fakeAddr2}
-
-		iface := &v1.Interface{Name: "default", InterfaceBindingMethod: v1.InterfaceBindingMethod{Masquerade: &v1.InterfaceMasquerade{}}}
-		mockNetwork.EXPECT().ReadIPAddressesFromLink(primaryPodInterfaceName).Return(fakeAddr1.IP.String(), fakeAddr2.IP.String(), nil)
-		mockNetwork.EXPECT().LinkByName(primaryPodInterfaceName).Return(primaryPodInterface, nil)
-		mockNetwork.EXPECT().AddrList(primaryPodInterface, netlink.FAMILY_ALL).Return(addrList, nil)
-		mockNetwork.EXPECT().IsIpv4Primary().Return(true, nil).Times(1)
-		podnic := newPodNIC(vmi)
-		podnic.iface = iface
-		podnic.podInterfaceName = primaryPodInterfaceName
-		err := podnic.setPodInterfaceCache()
-		Expect(err).ToNot(HaveOccurred())
-
-		podData, err := cacheFactory.CacheForVMI(vmi).Read(iface.Name)
-		Expect(err).ToNot(HaveOccurred())
-		Expect(podData.PodIP).To(Equal("1.2.3.4"))
```

Comment:
no, the test is testing the use of the cache inside the podnic, so it's apodnic test.

Reply:
I guess what I'm really trying to say is it feels you're testing a "private" method of the pod nic - i.e. an implementation detail.

I guess since golang doesn't have that concept, I might be wrong. And it already existed.

Reply:
well this is a unit test testing private method is usually done.